### PR TITLE
Cicada pattern testing

### PR DESCRIPTION
**Pull request by aloeliger**
### PR description:

This PR adds a way to configure the CICADA emulator to run an arbitrary input pattern to test accuracy against firmware/hardware implementations of CICADA. This testing is done in situ, running the actual emulator in CMSSW by overwriting the inputs of an otherwise valid event to provide CICADA outputs (A warning about this behavior is issued when activated). This behavior is, of course, disabled by default. Messages are provided at the Info level.

This behavior should never be used in typical CMSSW operation, this is intended for CICADA team validation of the genuine emulator in as close to real scenarios as possible to ensure that it is functioning as intended.

### PR validation:

This PR has been tested on 3 zero bias events, and it successfully runs CICADA on provided patterns. This pattern testing has been used to catch a firmware/emulator mismatch (another correcting PR is open for that). The test patterns used for validation in this test have been provided as `CICADATestPatterns.py`

### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This PR is not a backport, and is not intended to be backported.

### CODE REVIEW

**Comment by aloeliger** [comment]
please test

**Comment by aloeliger** [comment]
Test _numerous_ failures are due to a badly formatted import. It's fixable. Just embarrassing.

**Comment by aloeliger** [comment]
please test

**Comment by aloeliger** [comment]
+l1

File: L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc [hunk 1]
```diff
@@ -215,6 +223,32 @@ void L1TCaloSummary<INPUT, OUTPUT>::produce(edm::Event& iEvent, const edm::Event
...
+    unsigned int evt = iEvent.id().event();
+    unsigned int totalTestPatterns = testPatterns.size();
+    unsigned int patternElement = evt % totalTestPatterns;
+    const edm::ParameterSet& element = testPatterns.at(patternElement);
+    std::stringstream inputStream;
+    std::string PhiRowString;
+
+    edm::LogWarning("L1TCaloSummary") << "Overwriting existing CICADA input with test pattern!\n";
+
+    for (unsigned short int iPhi = 1; iPhi <= 18; ++iPhi) {
```
Comment (rappoccio) [inline L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc:237]: I'm a bit confused, why have this 1-indexed but then subtract the one when you access the array?

Comment (aloeliger) [reply L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc:237]: @antoniovilela @rappoccio Apologies, this fell off my radar as less important over the past few weeks. The number was done this way originally to line it up to the physical phi value as a convenience when I wrote it, but I can clean that up to be a little more appropriate if you prefer.

**Comment by rappoccio** [comment]
+1

**Pull request merged**